### DISCUSSION TITLE
How does the Complete node work?

### DISCUSSION THREAD

**Discussion by phoddie** [phoddie/node-red-mcu#11] [Q&A]
I don't really understand the `complete` node. I'd like to, so I can try to implement it. The documentation explains:

> Trigger a flow when another node completes its handling of a message.

That seems similar to `catch` and `status` nodes. But, I'm not sure how a node indicates it is complete. The documentation notes...

> Not all nodes will trigger this event - it will depend on whether they have been implemented to support this feature

... but doesn't explain which nodes do trigger a complete. It mentions the "Email sending node" supports complete, but do any of the core built-in Node-RED nodes support it?

**Comment by mtoko** [comment]
I would suggest joining the Node Red forum and the Slack channel and posting these types of questions. There are many knowledgeable people and the core Node Red team also monitor these  
https://discourse.nodered.org
https://nodered.org/about/community/slack/

**Comment by phoddie** [comment]
Thanks for the good suggestion. I've [posted a version of this question](https://discourse.nodered.org/t/how-does-the-complete-node-work/64733) to the Node-RED discourse server.
